src/app.tsx

```diff
@@ -1,18 +1,28 @@
 import { JSX } from 'react';
 import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import { store } from './store/store';
 import UserLogin from './components/UserLogin';
 import Homepage from './components/home-page';
+import LoginButton from './components/LoginButton';
+import LogoutButton from './components/LogoutButton';
 
 /**
  * The main application component.
  */
 export function App(): JSX.Element {
   return (
-    <Router>
-      <Routes>
-        <Route path="/" element={<Homepage />} />
-        <Route path="/login" element={<UserLogin />} />
-      </Routes>
-    </Router>
+    <Provider store={store}>
+      <Router>
+        <div>
+          <LoginButton />
+          <LogoutButton />
+        </div>
```

Comment:
The LoginButton and LogoutButton are rendered outside of routes in app.tsx. This means they'll be visible on all pages.
Please @Arielpetit, consider conditionally rendering the buttons based on the isLoggedIn state.